src/mongo_client.ts

```diff
@@ -34,16 +34,16 @@ import type { Encrypter } from './encrypter';
 import { TypedEventEmitter } from './mongo_types';
 
 /** @public */
-export const ServerApiVersion = {
+export const ServerApiVersion = Object.freeze({
   v1: '1'
-};
+} as const);
 
 /** @public */
-export type ServerApiVersionId = typeof ServerApiVersion[keyof typeof ServerApiVersion];
+export type ServerApiVersion = typeof ServerApiVersion[keyof typeof ServerApiVersion];
 
 /** @public */
 export interface ServerApi {
-  version: string | ServerApiVersionId;
+  version: string | ServerApiVersion;
```

Comment:
@emadum Do you happen to know why this is typed as `string | ServerApiVersion`? The api version is also a string, so this typing is equivalent to just `string` - do we actually want to allow arbitrary strings here?

Reply:
I don't recall a reason, we don't want to allow arbitrary strings so I think `string` can be removed.

Reply:
Actually, it looks like it's currently assigned when we parse the connection options on the condition that it is a string... do we want to restrict the option validation so that only the api versions known to the driver are supported? or do we want users to be able to specify any version regardless of whether it's known to the driver?

Reply:
Cross referencing the spec:
it looks like the `string | ServerApiVersion` given in the spec for the class is intended to accommodate languages that can't provide an enum; since we can provide an enum via typescript, it's fine for us to just have the enum, however in light of:
>  the driver MUST validate (e.g. when the application provides a version string to the ServerApi class) that the version string is valid and trigger a client-side error if an unknown API version was used

I'm going to add the check and error in the connection parser since typescript doesn't provide run time validation